nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/CandidateRepository.java

```diff
@@ -150,14 +167,28 @@ public void create(CandidateDao candidate, Collection<ApprovalStatusDao> approva
     addNewItemWithVersion(transaction, uniquenessTable, uniquenessEntry);
     addNewItemWithVersion(transaction, candidateTable, candidate);
 
+    if (nonNull(dependentPeriod)) {
+      transaction.addConditionCheck(periodTable, requireExpectedPeriodRevision(dependentPeriod));
+    }
```

Comment:
Note: This allows callers to pass `null` instead of the actual period a Candidate belongs to. This is done for two reasons:

1. To avoid rewriting `CristinNviReportEventConsumer` which migrates historical data from Cristin. Adding the dependent period to calls from this would both slow it down and require rewriting code that will soon be irrelevant.
2. To avoid rewriting irrelevant test cases where the relevant period is not easily available.